src/client/datascience/interactive-common/notebookServerProvider.ts

```diff
@@ -90,14 +83,29 @@ export class NotebookServerProvider implements IJupyterServerProvider {
             serverOptions.uri = await this.serverUriStorage.getUri();
         }
 
+        const disposables: IDisposable[] = [];
         // Status depends upon if we're about to connect to existing server or not.
-        const progressReporter =
+        let progressReporter =
             this.ui.disableUI === false
                 ? (await this.jupyterExecution.getServer(serverOptions))
                     ? this.progressReporter.createProgressIndicator(localize.DataScience.connectingToJupyter())
                     : this.progressReporter.createProgressIndicator(localize.DataScience.startingJupyter())
                 : undefined;
-
+        if (progressReporter) {
+            disposables.push(progressReporter);
+        } else {
+            this.ui.onDidChangeDisableUI(
+                async () => {
+                    if (!progressReporter && !this.ui.disableUI) {
+                        progressReporter = (await this.jupyterExecution.getServer(serverOptions))
```

Comment:
This is the magic sauce that re-displays the progress indicator as required.
I.e. if we have auto start, and it takes say 5-30 seconds or longer, and you start running a cell, now you'll immediately get a progress indicator as though you started the kernel from scratch